src/setup_node_env/node_version_validator.js

```diff
@@ -28,27 +28,201 @@
  * under the License.
  */
 
+/* Note:
+ *   This file uses ES5 in order to provide meaningful output even if an old Node.js runtime is used.
+ */
+
 var pkg = require('../../package.json');
 
-// Note: This is written in ES5 so we can run this before anything else
-// and gives support for older NodeJS versions
-var currentVersion = (process && process.version) || null;
-var rawRequiredVersion = (pkg && pkg.engines && pkg.engines.node) || null;
-var requiredVersion = rawRequiredVersion ? 'v' + rawRequiredVersion : rawRequiredVersion;
-var currentVersionMajorMinorPatch = currentVersion.match(/^v(\d+)\.(\d+)\.(\d+)/);
-var requiredVersionMajorMinorPatch = requiredVersion.match(/^v(\d+)\.(\d+)\.(\d+)/);
-var isVersionValid =
-  currentVersionMajorMinorPatch[1] === requiredVersionMajorMinorPatch[1] &&
-  currentVersionMajorMinorPatch[2] === requiredVersionMajorMinorPatch[2] &&
-  parseInt(currentVersionMajorMinorPatch[3], 10) >= parseInt(requiredVersionMajorMinorPatch[3], 10);
-
-// Validates current the NodeJS version compatibility when OpenSearch Dashboards starts.
-if (!isVersionValid) {
-  var errorMessage =
-    `OpenSearch Dashboards was built with ${requiredVersion} and does not support the current Node.js version ${currentVersion}. ` +
-    `Please use Node.js ${requiredVersion} or a higher patch version.`;
-
-  // Actions to apply when validation fails: error report + exit.
-  console.error(errorMessage);
-  process.exit(1);
+var pkgEngineNodeVersion = pkg && pkg.engines && pkg.engines.node;
+if (!pkgEngineNodeVersion || typeof pkgEngineNodeVersion !== 'string') {
+  console.error(
+    'OpenSearch Dashboards did not report its required version of the Node.js runtime. ' +
+      'Please revert any changes that might have been made to the package.json file and try again.'
+  );
+  process.exit(22);
```

Comment:
what are these exit codes stands for? Thought the commonly used ones are 0-12 or >128 🧐

Reply:
The numbers have no significance outside of OSD. 1, 2, 126 - 137, and 255 are used by the system and perhaps best to avoid; the rest can be used as desired.

Reply:
I don't think we have any; we have a few `64` and `70`, and a ton of `1`. However, since these exit codes are meant ONLY to help tests, I donno if we should care to document them. I have no strong feelings and can easily be swayed.

Reply:
I documented them at the top of the file.